src/jsonpatcherproxy.js

```diff
@@ -6,88 +6,98 @@
  */
 
 /** Class representing a JS Object observer  */
-var JSONPatcherProxy = (function() {
-  function JSONPatcherProxy(root) {
+const JSONPatcherProxy = (function() {
+  const proxifiedObjectsMap = new Map();
+
+  /**
+    * Creates an instance of JSONPatcherProxy around your object of interest `root`. 
+    * @param {Object|Array} root - the object you want to wrap
+    * @param {Boolean} showDetachedWarning - whether to log a warning when a detached sub-object is modified @see {@link https://github.com/Palindrom/JSONPatcherProxy#detached-objects} 
```

Comment:
WDYT about adding optional parameter info and default values in jsdoc? http://usejsdoc.org/tags-param.html#optional-parameters-and-default-values